src/main/java/duke/command/Command.java

```diff
@@ -0,0 +1,40 @@
+package duke.command;
+
+import duke.DukeException;
+import duke.util.TaskList;
+import duke.util.Storage;
+import duke.util.Ui;
+
+/**
+ * Represents a user command and its functionalities.
+ */
+public abstract class Command {
```

Comment:
I really like your use of abstract classes to abstract the different types of commands!